fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/objects/security/KeyPairTest.kt

```diff
@@ -50,6 +53,27 @@ class KeyPairTest {
     Assert.assertEquals("SGnNfF533PBEUMYPMqBSQY83z5U=", pk.computeHash())
   }
 
+  @Test
+  fun pubKeyUsernameDigits() {
+    val pk = PublicKey("oKHk3AivbpNXk_SfFcHDaVHcCcY8IBfHE7auXJ7h4ms=")
+    val digits = "3877"
+    // last 4 characters of the hash are the 4 first numerical digits of the hash
+    Assert.assertEquals(digits, pk.getLabel().substring(pk.getLabel().length - USERNAME_DIGITS))
+  }
+
+  @Test
+  fun pubKeyUsernameHashContainsLessDigits() {
+    val pk = PublicKey("oKHk3AivbpNXk_SfFcHDaVHcCcY8IBfHE7auXJmhmms=")
+    val digits = "0387"
+    // If the Hash contains less than 4 digits, the username will be padded with 0
+    Assert.assertEquals(digits, pk.getLabel().substring(pk.getLabel().length - USERNAME_DIGITS))
+  }
+
+  @Test
```

Comment:
Maybe you could add one or two more tests then just the empty username for the generateUsername function